toml/decoder.py

```diff
@@ -957,3 +964,54 @@ def load_array(self, a):
                     atype = ntype
                 retval.append(nval)
         return retval
+
+    def handle_inline_comment(self, line, currentlevel):
+        return line
+
+    def handle_comment(self, currentlevel, comment_no, recent, line):
+        return comment_no
+
+    def comment_lint(self, sl, i, beginline):
+        j = i
+        try:
+            while sl[j] != '\n':
+                sl[j] = ' '
+                j += 1
+        except IndexError:
+            raise IndexError()
+        return beginline
+
+
+class TomlPreserveCommentDecoder(TomlDecoder):
+
+    def __init__(self, _dict=OrderedDict):
+        super(TomlPreserveCommentDecoder, self).__init__(_dict)
+
+    def handle_inline_comment(self, line, currentlevel):
+        pos = line.find('#')
+        k = pos
+        comment = ""
+        try:
+            while line[k] != '\n':
+                comment += line[k]
+                k += 1
+        except IndexError:
+            pass
+        pair = line.split('=')
+        currentlevel[pair[0].strip() + '-' + 'comment' + '_inline'] = comment
+        line = line.split('#')[0]
+        return line
+
+    def handle_comment(self, currentlevel, comment_no, recent, line):
+        if recent + '-comment' in currentlevel:
+            currentlevel[recent + '-' + str(comment_no) + '-' + 'comment'] \
+                = line
+            comment_no += 1
+        else:
+            currentlevel[recent + '-' + 'comment'] = line
```

Comment:
Same thing as above but for keys ending in `-comment` which seems likely to me to be far more common...